drivers/charger/charger_max20335.c

```diff
@@ -0,0 +1,265 @@
+	uint8_t val;
+
+	switch (current_ua) {
+	case 0:
+		val = 0x00;
+		break;
+	case 100000:
+		val = 0x01;
+		break;
+	case 500000:
+		val = 0x02;
+		break;
+	case 1000000:
+		val = 0x03;
+		break;
+	default:
+		return -ENOTSUP;
+	};
+
+	val = FIELD_PREP(MAX20335_ILIMCNTL_MASK, val);
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      MAX20335_REG_ILIMCNTL,
+				      MAX20335_ILIMCNTL_MASK,
+				      val);
+}
+
+static int charger_max20335_get_constant_charge_current(const struct device *dev,
+							uint32_t *current_ua)
+{
+	const struct charger_max20335_config *const config = dev->config;
+	uint8_t val;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->bus, MAX20335_REG_ILIMCNTL, &val);
+	if (ret) {
+		return ret;
+	}
+
+	switch (val) {
+	case 0x00:
+		*current_ua = 0;
+		break;
+	case 0x01:
+		*current_ua = 100000;
+		break;
+	case 0x02:
+		*current_ua = 500000;
+		break;
+	case 0x03:
+		*current_ua = 1000000;
+		break;
+	default:
+		return -ENOTSUP;
+	};
+
+	return 0;
+}
+
+static int charger_max20335_get_constant_charge_voltage(const struct device *dev,
+							uint32_t *current_uv)
+{
+	const struct charger_max20335_config *const config = dev->config;
+	uint8_t val;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->bus, MAX20335_REG_CHG_CNTL_A, &val);
+	if (ret) {
+		return ret;
+	}
+
+	val = FIELD_GET(MAX20335_CHGCNTLA_BAT_REG_CFG_MASK, val);
+
+	return linear_range_get_value(&charger_uv_range, val, current_uv);
+}
+
+static int charger_max20335_set_enabled(const struct device *dev, bool enable)
+{
+	const struct charger_max20335_config *const config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus,
+				      MAX20335_REG_CHG_CNTL_A,
+				      MAX20335_CHRG_EN_MASK,
+				      enable ? MAX20335_CHRG_EN : 0);
+}
+
+static int charger_max20335_get_prop(const struct device *dev, charger_prop_t prop,
+				     union charger_propval *val)
+{
+	switch (prop) {
+	case CHARGER_PROP_STATUS:
+		return charger_max20335_get_status(dev, &val->status);
+	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
+		return charger_max20335_get_constant_charge_current(dev,
+								    &val->const_charge_current_ua);
+	case CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE_UV:
+		return charger_max20335_get_constant_charge_voltage(dev,
+								    &val->const_charge_voltage_uv);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int charger_max20335_set_prop(const struct device *dev, charger_prop_t prop,
+				     const union charger_propval *val)
+{
+	switch (prop) {
+	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
+		return charger_max20335_set_constant_charge_current(dev,
+								    val->const_charge_current_ua);
+	case CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE_UV:
+		return charger_max20335_set_constant_charge_voltage(dev,
+								    val->const_charge_voltage_uv);
+	case CHARGER_STATUS_CHARGING:
+		return charger_max20335_set_enabled(dev, true);
+	case CHARGER_STATUS_NOT_CHARGING:
+		return charger_max20335_set_enabled(dev, false);
```

Comment:
You are not setting a property here.
```c
case CHARGER_PROP_STATUS:
    if (val.status == CHARGER_STATUS_CHARGING) {
        return charger_max20335_set_enabled(dev, true);
    } else {
        return charger_max20335_set_enabled(dev, false);
    }
```
Something like this would be what you're trying to do.

Reply:
Oops, you are absolutely right. I was using that in the wrong way.